### PR TITLE
Adds loading indicators on weather buttons

### PR DESCRIPTION
**Pull request by madsnedergaard**
## Description

Since loading weather data and layers can take some time, it's hard to know as a user that something is happening. This should help solve that :) The buttons also had so much similar logic that I decided to abstract it - dunno if worth it, but now it's done ¯\\\_(ツ)\_/¯

### Preview

Kapture 2022-12-20 at 23 23 24

### CODE REVIEW

File: web/src/features/map-controls/MapControls.tsx [hunk 1]
```diff
@@ -4,30 +4,72 @@ import { ReactElement, useState } from 'react';
...
+          <Icon size={weatherButtonMap[type].iconSize} color={isEnabled ? '' : 'gray'} />
+        )
+      }
+      tooltipText={__(`tooltips.${type}}`)}
+      dataTestId={`${type}-layer-button`}
+      className={`mt-2 ${isLoadingLayer ? 'cursor-default' : 'cursor-pointer'}`}
+      onClick={!isLoadingLayer ? onToggle : () => {}}
+    />
+  );
+}
```
Comment (Kongkille) [inline web/src/features/map-controls/MapControls.tsx:65]: Should the mapbutton be disabled while loading?

File: web/src/features/map-controls/MapControls.tsx [hunk 2]
```diff
@@ -4,30 +4,72 @@ import { ReactElement, useState } from 'react';
...
+
+export default function MapControls(): ReactElement {
+  const { __ } = useTranslation();
+  const [isLanguageSelectorOpen, setIsLanguageSelectorOpen] = useState(false);
+  const [selectedDatetime] = useAtom(selectedDatetimeIndexAtom);
+
+  const areWeatherLayersAllowed = selectedDatetime.index === 24;
```
Comment (Kongkille) [inline web/src/features/map-controls/MapControls.tsx:72]: Perhaps a comment here explaining it's only available on latest value?

**Review by Kongkille — APPROVED** [review]

File: web/src/features/map-controls/MapControls.tsx (hunk 1 above)
Comment (madsnedergaard) [reply web/src/features/map-controls/MapControls.tsx:65]: It is with this: `onClick={!isLoadingLayer ? onToggle : () => {}}`
I guess we could use a "disabled" state instead, but doesn't change much :)

File: web/src/features/map-controls/MapControls.tsx (hunk 2 above)
Comment (madsnedergaard) [reply web/src/features/map-controls/MapControls.tsx:72]: Good idea, any suggestions to text? 😅

Comment (Kongkille) [reply web/src/features/map-controls/MapControls.tsx:72]: We are currently only supporting and fetching weather data for the latest hourly value

File: web/src/features/map-controls/MapControls.tsx (hunk 1 above)
Comment (Kongkille) [reply web/src/features/map-controls/MapControls.tsx:65]: Ah yeah ok :D

File: web/src/features/map-controls/MapControls.tsx (hunk 2 above)
Comment (madsnedergaard) [reply web/src/features/map-controls/MapControls.tsx:72]: Good points, done! :)

**Pull request merged**